scipy/optimize/_lsq/least_squares.py

```diff
@@ -4,10 +4,11 @@
 import numpy as np
 from numpy.linalg import norm
 
-from scipy.sparse import issparse, csr_matrix
```

Comment:
I'm not sure why `csr_matrix` import is needed in this PR?

Reply:
I removed the import of `csr_matrix` as it was unused.